### PR TITLE
Feature/Soul-Unlocks-parse

### PR DESCRIPTION
**Pull request by Surxe**
Parses soul unlock related data.

Data is uploaded to https://deadlocked.wiki/Data:SoulUnlockData.json

Module is at https://deadlocked.wiki/Module:SoulUnlock

Module currently supports two structures, the one uploaded, and an aggregated structure. Has localization support too so it all works nice 'n dandy already :)

### CODE REVIEW

File: src/parser/maps.py [hunk 1]
```diff
@@ -20,6 +20,14 @@ def get_target_type(value):
     return TARGET_TYPE_MAP.get(value)
 
 
+SOUL_UNLOCK_MAP = {
+    'm_unRequiredGold': 'RequiredGold',
```
Comment (hassayag) [inline src/parser/maps.py:24]: Is there a good reason to not call this souls?

File: src/parser/parsers/souls.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
+from parser.maps import SOUL_UNLOCK_MAP
+
+
+class SoulUnlockParser:
```
Comment (hassayag) [inline src/parser/parsers/souls.py:4]: Brief description of what soul unlock means would be good

File: src/parser/parsers/souls.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                if parsed_soul_unlocks == {}:
+                    parsed_soul_unlocks = hero_soul_unlocks
+                    hero_key_of_parsed_soul_unlocks = hero_key
+                # Confirm that the soul unlocks is the same as the previous hero's soul unlocks
+                else:
+                    if parsed_soul_unlocks != hero_soul_unlocks:
+                        raise ValueError(
+                            'Soul unlocks do not match between heroes'
+                            + f'{hero_key_of_parsed_soul_unlocks} and {hero_key}'
+                        )
```
Comment (hassayag) [inline src/parser/parsers/souls.py:28]: ah very nice, i just had this thought pop up, since they show them per hero in game

**Review by hassayag — CHANGES_REQUESTED** [review]
Very nice, future proofs us for inevitable changes

File: src/parser/maps.py (hunk 1 above)
Comment (Surxe) [reply src/parser/maps.py:24]: Changed to RequiredSouls on backend and frontend

File: src/parser/parsers/souls.py (hunk 2 above)
Comment (Surxe) [reply src/parser/parsers/souls.py:4]: Good point, added!

**Review by hassayag — APPROVED** [review]

**Pull request merged**